BDArmory/Ammo/PooledBullet.cs

```diff
@@ -1664,8 +1664,14 @@ public void BeehiveDetonation()
                 float incrementVelocity = 1000 / (bulletVelocity + sBullet.bulletVelocity); //using 1km/s as a reference Unit
                 float dispersionAngle = sBullet.subProjectileDispersion > 0 ? sBullet.subProjectileDispersion : BDAMath.Sqrt(count) / 2; //fewer fragments/pellets are going to be larger-> move slower, less dispersion
                 float dispersionVelocityforAngle = 1000 / incrementVelocity * Mathf.Sin(dispersionAngle * Mathf.Deg2Rad); // convert m/s despersion to angle, accounting for vel of round
+               
                 float subProjVelocity = GetDragAdjustedVelocity().magnitude + sBullet.bulletVelocity;
-                float subDetonationRange = BlastPhysicsUtils.CalculateBlastRange(sBullet.tntMass) * (sBullet.nuclear ? 2f : 0.666f); // The default for BDExplosivePart, unless also nuclear, in which case double the tnt blast radius? FIXME What's a better detonation range for these?
+                float subDetonationRange = 0;
+                bool isSabot = ((sBullet.bulletMass * 1000 / (sBullet.caliber * sBullet.caliber * Mathf.PI / 400 * 19) + 1) * 10) > sBullet.caliber * 4; //moving these here do they don't need to be re-calced 20 times per beehive det
+                if (sBullet.tntMass > 0)
+                {
+                    subDetonationRange = sBullet.nuclear ? Mathf.Pow(sBullet.tntMass, 0.33333f) * 10 * (10 * atmosphereDensity) : BlastPhysicsUtils.CalculateBlastRange(sBullet.tntMass) * 0.666f;
```

Comment:
very rough rule of thumb approximation, but sufficient for the sorts of yields you'd expect from nuclear shells

Reply:
For very low yield ones, yes. Even in atmo very low yield nukes aren't going to do much against armored vehicles.
In the coilgun's case, it's also a config issue; the 6kg mass of the submunitions means there simply isn't that much mass there in the first place to get turned into plasma, which isn't helping either. Kick that up to a more reasonable 30-40kg per, and they'd start doing substantially more damage.
From the reports in testing, the primary threat seemed to be from EMP and bricking the target vessel.